akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpHeader.scala

```diff
@@ -16,6 +16,19 @@ import akka.util.OptionVal
 
 import scala.collection.immutable
 
+/**
+ * Marker trait for headers which contain portentially secret / sensitive information.
+ *
+ * Mixing this trait will make `toString` to return the name of the header thus avoiding any
+ * detail leak.
+ */
+trait SensitiveHttpHeader {
+  this: HttpHeader =>
+
+  // This header is tagged as potentially containing personal sensitive information
+  override def toString: String = name
```

Comment:
That this works is somewhat subtle given that below `def unsafeToString` is implemented in terms of `super.toString`. Should we mark this method as `final` to ensure it is the definitive implementation of `toString` in cases where it is used?

Reply:
Indeed, I was about to open a PR to include your suggestion just to find that you already did (https://github.com/jrudolph/akka-http/commit/626485284a0b661f487418dee8284eb2c789ff90). Thank you :bow:

Reply:
I've also opened another follow-up PR cleaning the outdated exclusion added here: https://github.com/akka/akka-http/pull/3181